### PR TITLE
Align configuration with OpenTelemetry SDK, HTTP exporter

### PR DESCRIPTION
**Pull request by olegbespalov**
# What?

This PR aligns configuration (k6 environment variables) with the OpenTelemetry SDK and introduces more configuration options, along with the HTTP metric exporter.

# Why?

It's essential to be aligned so as not to confuse the users, as proven [in the comment](https://github.com/grafana/xk6-output-opentelemetry/issues/1#issuecomment-2123478609).

### CODE REVIEW

**Comment by cwegener** [comment]
Looks good to me! :+1:

File: pkg/opentelemetry/config.go [hunk 1]
```diff
@@ -5,82 +5,186 @@ import (
...
-	ReceiverType string
-	// GRPCReceiverEndpoint is the endpoint of the gRPC receiver
-	GRPCReceiverEndpoint string
-	// PushInterval is the interval at which to push metrics to the receiver
-	PushInterval time.Duration
 	// FlushInterval is the interval at which to flush metrics from the k6
 	FlushInterval time.Duration
+
+	// ExporterType sets the type of OpenTelemetry Exporter to use
+	// Currently only "grpc" is supported
```
Comment (joanlopez) [inline pkg/opentelemetry/config.go]: Don't we support `http`?  🤔 If not, why are all the HTTP references used for?

Comment (olegbespalov) [reply pkg/opentelemetry/config.go]: Good catch :+1: It's a leftover, will remove this line :+1:

Comment (joanlopez) [reply pkg/opentelemetry/config.go]: I think the line in the validation method confirms it: 

```
if c.ExporterType != grpcExporterType && c.ExporterType != httpExporterType {
```

So I guess we can just remove that old comment 😛

File: pkg/opentelemetry/config.go [hunk 2]
```diff
@@ -5,82 +5,186 @@ import (
...
 
-	if c.GRPCReceiverEndpoint == "" {
-		return errors.New("gRPC receiver endpoint is required")
+	if c.ServiceVersion == "" {
+		return errors.New("providing service version is required")
+	}
+
+	if c.ExporterType != grpcExporterType && c.ExporterType != httpExporterType {
+		return fmt.Errorf(
+			"unsupported exporter type %q, currently only %q and %q supported",
```
Comment (joanlopez) [inline pkg/opentelemetry/config.go]:
```suggestion
			"unsupported exporter type %q, currently only %q and %q are supported",
```

File: pkg/opentelemetry/config_test.go [hunk 3]
```diff
@@ -21,31 +25,43 @@ func TestConfig(t *testing.T) {
...
-			env: map[string]string{"K6_OTEL_GRPC_RECEIVER_ENDPOINT": "else", "K6_OTEL_PUSH_INTERVAL": "4something"},
+			env: map[string]string{"K6_OTEL_GRPC_EXPORTER_ENDPOINT": "else", "K6_OTEL_EXPORT_INTERVAL": "4something"},
 			err: `time: unknown unit "something" in duration "4something"`,
 		},
 
 		"unsupported receiver type": {
-			env: map[string]string{"K6_OTEL_GRPC_RECEIVER_ENDPOINT": "else", "K6_OTEL_PUSH_INTERVAL": "4m", "K6_OTEL_RECEIVER_TYPE": "http"},
-			err: `error validating config: unsupported receiver type "http", currently only "grpc" supported`,
+			env: map[string]string{"K6_OTEL_GRPC_EXPORTER_ENDPOINT": "else", "K6_OTEL_EXPORT_INTERVAL": "4m", "K6_OTEL_EXPORTER_TYPE": "socket"},
+			err: `error validating OpenTelemetry output config: unsupported exporter type "socket", currently only "grpc" and "http" supported`,
```
Comment (joanlopez) [inline pkg/opentelemetry/config_test.go]:
```suggestion
			err: `error validating OpenTelemetry output config: unsupported exporter type "socket", currently only "grpc" and "http" are supported`,
```

**Review by joanlopez — APPROVED** [review]

File: pkg/opentelemetry/config.go (hunk 1 above)
Comment (olegbespalov) [inline pkg/opentelemetry/config.go]:
```suggestion
```

**Review by joanlopez — APPROVED** [review]
I've added a few minor comments but overall looks great! 🚀

File: pkg/opentelemetry/config.go [hunk 4]
```diff
@@ -5,82 +5,185 @@ import (
...
+
+		HTTPExporterInsecure: null.BoolFrom(false),
+		HTTPExporterEndpoint: "localhost:4318",
+		HTTPExporterURLPath:  "/v1/metrics",
+
+		GRPCExporterInsecure: null.BoolFrom(false),
+		GRPCExporterEndpoint: "localhost:4317",
+
+		ExportInterval: 1 * time.Second,
+		FlushInterval:  1 * time.Second,
```
Comment (mstoykov) [inline pkg/opentelemetry/config.go:71]: nit: In general it is better idea to always use `null` types for everything as that lets you know if you should use a default value or if somebody just managed to match it.

File: pkg/opentelemetry/config.go [hunk 5]
```diff
@@ -5,82 +5,185 @@ import (
...
 		switch k {
-		case "K6_OTEL_PUSH_INTERVAL":
-			cfg.PushInterval, err = time.ParseDuration(v)
-			if err != nil {
-				return cfg, fmt.Errorf("error parsing environment variable 'K6_OTEL_PUSH_INTERVAL': %w", err)
-			}
+		case "K6_OTEL_SERVICE_NAME":
+			cfg.ServiceName = v
+		case "K6_OTEL_SERVICE_VERSION":
+			cfg.ServiceVersion = v
```
Comment (mstoykov) [inline pkg/opentelemetry/config.go:80]: Not certain if jut using envconfig.Process as all the other outputs will not be better especially before we move the whole configuration over to not using envconfig.

**Review by mstoykov — APPROVED** [review]
LGTM!

I left two comments which can be addressed later as well - but might be good idea to happen before we put this in core.

**Comment by olegbespalov** [comment]
@mstoykov, both comments are relevant. I'll try to implement the following configuration-related PRs. This is mostly about aligning with OpenTelemetry terms, so will merge this as it is for now :+1:

File: pkg/opentelemetry/config.go (hunk 4 above)
Comment (olegbespalov) [reply pkg/opentelemetry/config.go:71]: Implemented as part of: #11

File: pkg/opentelemetry/config.go (hunk 5 above)
Comment (olegbespalov) [reply pkg/opentelemetry/config.go:80]: Also, implemented as part of: #11

**Pull request merged**